File: rest-api/src/main/java/org/brooklyncentral/catalog/dto/Repository.java
package org.brooklyncentral.catalog.dto;

public class Repository {

    private final String repoUrl;
    private final String repoName;
    private final String author;
    private final String token;

    public Repository(String repoUrl, String repoName, String author) {
        this.repoUrl = repoUrl;
        this.repoName = repoName;
        this.author = author;
        this.token = author + "/" + repoName;
    }

    public String getRepoUrl() {
        return repoUrl;
    }

    public String getRepoName() {
        return repoName;
    }

    public String getAuthor() {
        return author;
    }

    public String getToken() {
        return token;
    }
}

File: jsgui/src/main/app/assets/app/model/repository.js
define([
    'underscore', 'backbone'
], function (_, Backbone) {

    var Repository = {};

    Repository.Model = Backbone.Model.extend({
       defaults: function() {
           return {
               repoUrl: null,
               repoName: null,
               author: null,
               token: null
           }
       }
    });

    Repository.Collection = Backbone.Collection.extend({
        model: Repository.Model,
        url: 'api/repositories'
    });

    return Repository;
});